File: algorithms/cpp/326.cpp
#include <iostream>
#include <cmath>
using namespace std;

class Solution
{
public:
    bool isPowerOfThree(int n)
    {
        double logResule = log(n)/log(3);
        if ( abs(logResule-(int)(logResule+0.5)) <= 1e-10 )
            return true;
        else
            return false;
    }
};

int main()
{
    Solution solution;
    solution.isPowerOfThree(243);
    return 0;
}

File: algorithms/python/326.py
from math import log
class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0: return False
        t = int(log(n, 3)+0.5)
        return  3**t == n
